src/OgGroupManager.php

```diff
@@ -0,0 +1,137 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\og\OgGroupManager.
+ */
+
+namespace Drupal\og;
+
+use Drupal\Core\Config\ConfigFactoryInterface;
+
+/**
+ * A manager to keep track of which entity type/bundles are OG group enabled.
+ */
+class OgGroupManager {
+
+  /**
+   * The config instance.
+   *
+   * @var \Drupal\Core\Config\Config
+   */
+  protected $config;
+
+  /**
+   * The config factory.
+   *
+   * @var \Drupal\Core\Config\ConfigFactoryInterface
+   */
+  protected $configFactory;
+
+  /**
+   * A map of entity types and bundles.
+   *
+   * @var array
+   */
+  protected $groupMap;
+
+  /**
+   * Constructs an OgGroupManager object.
+   */
+  public function __construct(ConfigFactoryInterface $config_factory) {
+    $this->config = $config_factory->get('og.settings');
+    $this->configFactory = $config_factory;
+  }
+
+  /**
+   * Determines whether an entity type ID and bundle ID are group enabled.
+   *
+   * @param string $entity_type_id
+   * @param string $bundle
+   *
+   * @return bool
+   */
+  public function isGroup($entity_type_id, $bundle) {
+    if (!isset($this->groupMap)) {
```

Comment:
this IF part that is duplicated bugs me a bit, it implies we don't have full control over it's state. can't we move it to the constructor?